Jenkinsfile

```diff
@@ -7,20 +7,35 @@ node {
             script: 'make version',
             returnStdout: true
         ).trim()
+        def props = [ version: version ]
 
         stage("Testing ${version}") {
             sh 'make test-docker'
         }
         stage("Build image ${version}") {
             sh 'make build-docker'
         }
-
         if (env.BRANCH_NAME == "master") {
             stage("Push image") {
                 sh 'make push publish'
             }
+        } else {
+            def branchMatch = env.BRANCH_NAME =~ /^release-(0|[1-9]\d*)\.(0|[1-9]\d*)\.x$/
+            if (branchMatch) {
+                def releaseVersion = readFile "${env.WORKSPACE}/operator-version"
+                def versionMatch = releaseVersion =~ /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(.+))?$+/
+                if (!versionMatch) {
+                    error("Incorrect version ${releaseVersion} defined in ./operator-version")
+                }
+                if (versionMatch.group(1) != branchMatch.group(1) || versionMatch.group(2) != branchMatch.group(2)) {
+                    error("Version ${releaseVersion} does not match branch it is being built on ${env.BRANCH_NAME}")
+                }
+                props.releaseVersion = releaseVersion
+                stage("Publish Version ${releaseVersion}") {
+                    sh "make push publishRelease RELEASE_VERSION=\"${releaseVersion}\""
```

Comment:
hmmmmmmmm do we have to commit the `operator-version`? why not just pull it from the branch?

Reply:
For patch releases.